### ISSUE TITLE
Location doesn't work on Android

### ISSUE THREAD

**Issue by M-HARIS-97** [5-stones/react-native-readium#16]
Location doesn't work on Android. Always opens the first page. Locator props works fine.

<ReadiumView
          file={{url: filePath}}
          settings={settings}
          location={{
            href: '/EPUB/text/ch002.xhtml',
            locations: {
              position: 5,
              progression: 0,
              totalProgression: 0.024390243902439025,
            },
            type: 'application/xhtml+xml',
          }}
          onLocationChange={locator => locationChanged(locator)}
        />

Error: lateinit property navigator has not been initialized

**Comment by jspizziri** [comment]
@M-HARIS-97 interesting, this was working before. I'll try to take a look at it today. A full stacktrace from the android side would be helpful.

**Comment by jspizziri** [comment]
@M-HARIS-97 , I just tested this in the `example` app and I'm unable to reproduce it.

Can you please pull the latest version of the project and see if you can reproduce it in the example app? [I've added instructions on how to run the example app](https://github.com/5-stones/react-native-readium/tree/main/example#building--running).

You can test it by selecting one of the chapters from the table of contents. You can see that the navigator correctly navigates to the chapter. I'm going to close this until you figure out a repro in the example.

**Comment by jspizziri** [comment]
@M-HARIS-97 you need to describe to me how to reproduce the error in the example app. Have you tried that?

**Comment by M-HARIS-97** [comment]
@jspizziri My application has 2 screens. First is an epub file list screen and second is a readium reading screen. I select an epub file, flick through some pages and store the last page locator value in my local db (sqlite), then return to the listing page. I select the same book again, that time I retrieve the locator value from the local DB and pass the value to the location property. That time I got an error message which I shared earlier.

**Comment by jspizziri** [comment]
@M-HARIS-97 , I don't want to sound like a jerk. I'll help you solve this, but you need to create a reproduction in the example app. The problem could be many things, including your ebook. If I can't see a reproduction I won't be able to fix it or help you.

Do you have questions on how to set up a repro with the example?

**Comment by M-HARIS-97** [comment]
@jspizziri I was able to reproduce the problem using the example app. I made some changes to the example app. Please refer to the following to reproduce the issue;
1 - Install sqlite db
2 - Store the locator value in DB
3 - Close the app
4 - Then, open the app again
5 - Retrieve the locator value from DB
6 - Pass that value to the ReadiumView location parameter

**Comment by M-HARIS-97** [comment]
@jspizziri Please go through the following steps;

1 - Open the book. Flick some pages and console the locator value.
<img width="563" alt="Screenshot 2022-12-08 at 5 16 37 PM" src="https://user-images.githubusercontent.com/57429370/206438826-1c485b76-af3c-4638-84af-6ceadcadc972.png">
2 -  Copy that specific locator value from console.
3 - Then apply that locator value directly to the location props
<img width="425" alt="Screenshot 2022-12-08 at 5 17 38 PM" src="https://user-images.githubusercontent.com/57429370/206439034-b6202c30-b07f-44e9-9939-4786e9f8e8c0.png">
4 - Close the app
5 - Reopen the app
6 - **Expected** 
      6.1 - Opens the book and navigates to the specified location
      **But What I'm Seeing**
     6.2 - The book opens and shows its first page
     
**After updating to the latest version 1.0.4 its not working on iOS too.**

**Comment by M-HARIS-97** [comment]
@jspizziri I'm still facing this problem on iOS and android. Could you pleas verify this?

**Comment by jspizziri** [comment]
@M-HARIS-97 upon opening the App you should be using `initialLocation` which is part of the `File` object. The `location` attribute should really only be used during navigation once the reader view is booted. The reason for the duplicate property is to avoid a race condition. In order to properly build the view at the correct initial location the location needs to be guaranteed to be present at the time the view is built. Because of the asynchronous nature of the react-native bridge layer we can't guarantee that the `location` property will arrive at the native layer **before or at the same time** that the `file` property does. Therefore they need to be passed as the same object.

In short, `location` was never intended to be used for boot position, and it never should've been working on iOS or Android unless it was by pure luck that when you happened to run it, the async calls randomly happened to resolve in the correct order to make it work.

With all that said, I could see a potential need for a feature that reports the state of the views initialization process. Something like the following to help determine if you're allowed to start passing `locations`, etc: 

```ts
onStateChange={(state: 'loading' | 'initialized' | 'errored' ) => { ... }}
```

**Comment by M-HARIS-97** [comment]
@jspizziri Thanks for your detailed explanation. Currently it works well on iOS and android. Previously I was not using the **initialLocation** property. I think that's why I faced problems in my project. Once again Thanks @jspizziri for your great help.

**Comment by jspizziri** [comment]
@M-HARIS-97 thanks for the bug reports! I should probably add something to the README.md to make it more clear to folks what that property is for.

**Comment by jspizziri** [comment]
I've updated the readme for better clarity here https://github.com/5-stones/react-native-readium/commit/dc92e1e38c4c9f37aa8e7e078efba8501e2bdc91

**Comment by M-HARIS-97** [comment]
@jspizziri Ok Noted

**Comment by Darmolar** [comment]
I've tried using the initialLocation but am still getting the same result. Can you share a code snippet or how you initialized it?

**Comment by Darmolar** [comment]
Thank you @jspizziri. I was implementing it the wrong way.
